Code/GraphMol/new_canon.h

```diff
@@ -247,6 +245,37 @@ class RDKIT_GRAPHMOL_EXPORT SpecialSymmetryAtomCompareFunctor {
   }
 };
 
+namespace {
+unsigned int getChiralRank(const ROMol *dp_mol, canon_atom *dp_atoms,
+                           unsigned int i) {
+  unsigned int res = 0;
+  std::vector<unsigned int> perm;
+  perm.reserve(dp_atoms[i].atom->getDegree());
+  for (const auto nbr : dp_mol->atomNeighbors(dp_atoms[i].atom)) {
+    auto rnk = dp_atoms[nbr->getIdx()].index;
+    // make sure we don't have duplicate ranks
+    if (std::find(perm.begin(), perm.end(), rnk) != perm.end()) {
+      break;
+    } else {
+      perm.push_back(rnk);
+    }
+  }
+  if (perm.size() == dp_atoms[i].atom->getDegree()) {
+    auto ctag = dp_atoms[i].atom->getChiralTag();
+    if (ctag == Atom::ChiralType::CHI_TETRAHEDRAL_CW ||
+        ctag == Atom::ChiralType::CHI_TETRAHEDRAL_CCW) {
+      auto sortedPerm = perm;
+      std::sort(sortedPerm.begin(), sortedPerm.end());
+      auto nswaps = countSwapsToInterconvert(perm, sortedPerm);
```

Comment:
Another potential optimization, countSwapsToInterconvert could be made a sorting network (i.e. lookup)

Reply:
I'm not sure what you mean. Given that `countSwapsToInterconvert()` is called pretty frequently throughout the code, if you have an idea of how to make it faster a demo/PR would be great!